sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -985,6 +986,11 @@ def partial_fit(self):
 
     def _partial_fit(self, X, y, classes=None):
         _check_partial_fit_first_call(self, classes)
+        if not hasattr(self.label_binarizer_, "classes_"):
```

Comment:
This condition is not possible since the `classes_` attribute is set in `_check_partial_fit_first_call`. We should initialize a `LabelBinarizer` and fit with `self.classes_` if `_check_partial_fit_first_call` returns `True`.

Also why is the `LabelBinarizer` intialized in the constructor?